Prepare Redis Environment

JT2M0L3Y opened the pull request:
## Added
* redis container has been added to the docker infrastructure (both the compose and manual setup).

## Planned
* store setup and run-time configurations as key-value pairs in redis in the form of DB as a cache.

JT2M0L3Y commented:
![image](https://github.com/user-attachments/assets/a9c51a74-dd11-4ee8-851d-84bb97f869ff)
![image](https://github.com/user-attachments/assets/83527986-39e5-4096-8076-88a22b238ee5)

kevinthedang commented on package.json:
```diff
@@ -13,15 +13,16 @@
     "docker:ollama-cpu": "docker run -d -v ollama:/root/.ollama -p 11434:11434 --name ollama --network ollama-net --ip 172.18.0.2 ollama/ollama:latest",
-    "docker:start": "docker network prune -f && npm run docker:network && npm run docker:build && npm run docker:client && npm run docker:ollama",
-    "docker:start-cpu": "docker network prune -f && npm run docker:network && npm run docker:build && npm run docker:client && npm run docker:ollama-cpu"
+    "docker:start": "docker network prune -f && npm run docker:network && npm run docker:build && npm run docker:client && npm run docker:ollama && npm run docker:redis",
+    "docker:start-cpu": "docker network prune -f && npm run docker:network && npm run docker:build && npm run docker:client && npm run docker:ollama-cpu && npm run docker:redis"
```
Shouldn't we start the redis container before the discord application? Not sure how it will be initialized in the future but worth a thought.

JT2M0L3Y replied:
To store startup configuration info that may be needed by the discord app, yes.

kevinthedang reviewed (commented):
Looks great for now. Left a couple comments.

JT2M0L3Y commented:
## Latest Changes
* added connection to a Redis client
* updated npm packages, there were no big impediments (nothing used in this project was modified)

JT2M0L3Y commented:
## Changed
* added a requirement for npm versions based on the update to required node version
* updated node version used in containers

JT2M0L3Y commented:
Hoping to do a pre-release of the Redis environment here, this work is ready to be reviewed.

cc: @kevinthedang

kevinthedang commented on README.md:
```diff
@@ -27,12 +27,9 @@ The project aims to:
   * [x] Administrator Role Compatible
   * [x] Multi-User Chat Generation (Multiple users chatting at the same time) - This was built in from Ollama `v0.2.1+`
   * [x] Automatic and Manual model pulling through the Discord client
-* [ ] Allow others to create their own models personalized for their own servers!
-  * [ ] Documentation on creating your own LLM
-  * [ ] Documentation on web scrapping and cleaning
```
Are we totally scrapping this? We could probably create some documentation for this or link other work that does this.

JT2M0L3Y replied:
I was thinking so, I can drum up a quick doc on the Modelfile stuff for Ollama.

JT2M0L3Y replied:
I'm just going to link Ollama's docs on creating models, adding custom models, and the documentation for a `modelfile`.

kevinthedang commented on docs/commands-guide.md:
````diff
@@ -78,8 +78,7 @@ This is a guide to all of the slash commands for the app.
     ```
-    > [!NOTE]
-    > This is a very slow progress on Discord because "spamming" changes within 5 seconds is not allowed.
+    **This is a very slow progress on Discord because "spamming" changes within 5 seconds is not allowed.**
````
I'm criticizing my own sentence but probably "This is very slow on Discord because "spamming" changes in a channel within a period of 5 seconds is not allowed."

JT2M0L3Y replied:
Corrected.

kevinthedang commented on src/client.ts:
```diff
@@ -1,42 +1,55 @@
+// register all events
+registerEvents(client, Events, messageHistory, ollama, Keys.defaultModel)
+
+// Try to log in the client
+await client.login(Keys.clientToken)
+    .catch((error) => {
+        console.error('[Login Error]', error)
+        process.exit(1)
+    })
+
+// Try to connect to redis
+await redis.connect()
+    .then(() => console.log('[Redis] Connected'))
+    .catch((error) => {
+        console.error('[Redis] Connection Error', error)
+        process.exit(1)
+    })
+
+// queue up bots name
+messageHistory.enqueue({
+    role: 'assistant',
+    content: `My name is ${client.user?.username}`,
+    images: []
```
Not sure how all of this works, but should the order be:
1. Connect to Redis
2. Bring Bot Online

Not sure if there is anything that would cause issues if the bot was up and redis was still spinning. This, of course, is some random edge case I am thinking of that could probably not even be a problem.

JT2M0L3Y replied:
Order corrected, that makes more sense. Initially I had the redis.connect() right after creating the client (which was killing the process).

kevinthedang commented:
Also, it's chill to push multiple commits instead of force pushing some of the work to have the same commit name.
